aws-codeguruprofiler-profilinggroup/src/main/java/software/amazon/codeguruprofiler/profilinggroup/UpdateHandler.java

```diff
@@ -0,0 +1,109 @@
+import software.amazon.awssdk.services.codeguruprofiler.model.ActionGroup;
+import software.amazon.awssdk.services.codeguruprofiler.model.ConflictException;
+import software.amazon.awssdk.services.codeguruprofiler.model.GetPolicyRequest;
+import software.amazon.awssdk.services.codeguruprofiler.model.GetPolicyResponse;
+import software.amazon.awssdk.services.codeguruprofiler.model.InternalServerException;
+import software.amazon.awssdk.services.codeguruprofiler.model.PutPermissionRequest;
+import software.amazon.awssdk.services.codeguruprofiler.model.RemovePermissionRequest;
+import software.amazon.awssdk.services.codeguruprofiler.model.ResourceNotFoundException;
+import software.amazon.awssdk.services.codeguruprofiler.model.ThrottlingException;
+import software.amazon.awssdk.services.codeguruprofiler.model.ValidationException;
+import software.amazon.cloudformation.exceptions.CfnAlreadyExistsException;
+import software.amazon.cloudformation.exceptions.CfnInvalidRequestException;
+import software.amazon.cloudformation.exceptions.CfnNotFoundException;
+import software.amazon.cloudformation.exceptions.CfnServiceInternalErrorException;
+import software.amazon.cloudformation.exceptions.CfnThrottlingException;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+
+import java.util.List;
+
+public class UpdateHandler extends BaseHandler<CallbackContext> {
+
+    private final CodeGuruProfilerClient profilerClient = CodeGuruProfilerClientBuilder.create();
+
+    @Override
+    public ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+        AmazonWebServicesClientProxy proxy,
+        ResourceHandlerRequest<ResourceModel> request,
+        CallbackContext callbackContext,
+        Logger logger) {
+        final ResourceModel model = request.getDesiredResourceState();
+
+        final String awsAccountId = request.getAwsAccountId();
+        final String profilingGroupName = model.getProfilingGroupName();
+
+        Permissions newPermissions = model.getPermissions();
+
+        try {
+            GetPolicyResponse getPolicyResponse = getExistingPolicy(proxy, profilingGroupName);
+
+            if (newPermissions != null) {
+                List<String> principals = newPermissions.getAgentPermissions().getPrincipals();
+                putAgentPermissions(proxy, profilingGroupName, principals, getPolicyResponse.revisionId());
+                logger.log(
+                    String.format("Policy for [%s] for accountId [%s] has been successfully updated! actionGroup: %s, principals: %s",
+                        profilingGroupName,
+                        awsAccountId,
+                        ActionGroup.AGENT_PERMISSIONS,
+                        principals)
+                );
+            } else if(getPolicyResponse.policy() != null) {
```

Comment:
Here's a missing space after ```if```.

Reply:
That shows that I have not coded in Java for long ><!